sdk/tests/js/webgl-test-utils.js

```diff
@@ -3136,38 +3128,30 @@ var runSteps = function(steps) {
  * @param {!function(!HTMLVideoElement): void} callback Function to call when
  *        video is ready.
  */
-function startPlayingAndWaitForVideo(video, callback) {
+async function startPlayingAndWaitForVideo(video, callback) {
   if (video.error) {
     testFailed('Video failed to load: ' + video.error);
     return;
   }
 
-  video.addEventListener(
-      'error', e => { testFailed('Video playback failed: ' + e.message); },
-      true);
-
-  var rvfc = getRequestVidFrameCallback();
-  if (rvfc === undefined) {
-    var timeWatcher = function() {
-      if (video.currentTime > 0) {
-        callback(video);
-      } else {
-        requestAnimFrame.call(window, timeWatcher);
-      }
-    };
-
-    timeWatcher();
-  } else {
-    // Calls video.requestVideoFrameCallback(_ => { callback(video) })
-    rvfc.call(video, _ => { callback(video) });
-  }
-
   video.loop = true;
   video.muted = true;
   // See whether setting the preload flag de-flakes video-related tests.
   video.preload = 'auto';
-  video.play();
-};
+
+  try {
+    await video.play();
+  } catch (e) {
+    testFailed('video.play failed: ' + e);
+    return;
+  }
+
+  if (video.requestVideoFrameCallback) {
+    await new Promise(go => video.requestVideoFrameCallback(go));
```

Comment:
@dalecurtis do you think this will have the same basic semantics as the earlier code in Chromium?

Reply:
It will have different semantics as I note in the comment thread below, but it should be okay % potential for minor slowness.